### PR TITLE
DOCSP-34709: connection troubleshooting shared

### PR DESCRIPTION
**Pull request by ccho-mongodb**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-rust/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-34709>
Staging - [Connection Troubleshooting](https://preview-mongodbcchomongodb.gatsbyjs.io/rust/DOCSP-34709-connection-troubleshooting-shared/connection-troubleshooting/)

A few notable differences from the current page:
- Replaced duplicated "check connection string" section with guidance more relevant to each top-level topic
- Added a couple helpful links including links to the FAQ
- Reworded some of the headings

Vale issues responded to in comments.

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/connection-troubleshooting.txt [hunk 1]
```diff
@@ -10,198 +10,136 @@ Connection Troubleshooting
...
-.. _rust-troubleshooting-connection-string-auth:
+   .. replacement:: authentication-error-anchor
 
-Check Connection String
------------------------
+      .. _rust-troubleshooting-auth-error:
 
-An invalid connection string is the most common cause of authentication
-issues when attempting to connect to MongoDB.
+   .. replacement:: scram-failure-error
```
Comment (ccho-mongodb) [inline source/connection-troubleshooting.txt:88]: Cannot address this Vale issue, rule is set to change soon.

File: source/connection-troubleshooting.txt [hunk 2]
```diff
@@ -10,198 +10,136 @@ Connection Troubleshooting
...
 
-.. tip::
+      .. code-block:: none
+         :copyable: false
 
-   For more information about connection strings,
-   see the :ref:`rust-quick-start-connection-string` guide.
+          Error: Error { kind: Authentication { message: "SCRAM failure: bad auth :
```
Comment (ccho-mongodb) [inline source/connection-troubleshooting.txt:93]: Cannot address this Vale issue, rule is set to change soon.

File: source/connection-troubleshooting.txt [hunk 3]
```diff
@@ -10,198 +10,136 @@ Connection Troubleshooting
...
+          authentication failed" }, labels: {}, wire_version: None, source: Some(Error
```
Comment (ccho-mongodb) [inline source/connection-troubleshooting.txt:94]: Cannot address this Vale issue, rule is set to change soon.

File: source/connection-troubleshooting.txt [hunk 4]
```diff
@@ -10,198 +10,136 @@ Connection Troubleshooting
...
+          { kind: Command(CommandError { code: 8000, code_name: "AtlasError", message:
+          "bad auth : authentication failed", topology_version: None }),
```
Comment (ccho-mongodb) [inline source/connection-troubleshooting.txt:96]: Cannot address this Vale issue, rule is set to change soon.

**Review by rustagir — APPROVED** [review]
lgtm!

**Pull request merged**